src/Stratis.Bitcoin.Features.ColdStaking/Controllers/ColdStakingController.cs

```diff
@@ -0,0 +1,259 @@
+﻿using System;
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Features.ColdStaking.Models;
+using Stratis.Bitcoin.Features.Wallet;
+using Stratis.Bitcoin.Features.Wallet.Interfaces;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+using Stratis.Bitcoin.Utilities.ModelStateErrors;
+
+namespace Stratis.Bitcoin.Features.ColdStaking.Controllers
+{
+    /// <summary>
+    /// Controller providing operations for cold staking.
+    /// </summary>
+    [Route("api/[controller]")]
+    public class ColdStakingController : Controller
+    {
+        public ColdStakingManager ColdStakingManager { get; private set; }
+        private readonly IWalletTransactionHandler walletTransactionHandler;
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        public ColdStakingController(
+            ILoggerFactory loggerFactory,
+            IWalletManager walletManager,
+            IWalletTransactionHandler walletTransactionHandler)
+        {
+            Guard.NotNull(loggerFactory, nameof(loggerFactory));
+            Guard.NotNull(walletManager, nameof(walletManager));
+            Guard.NotNull(walletTransactionHandler, nameof(walletTransactionHandler));
+
+            this.ColdStakingManager = walletManager as ColdStakingManager;
+            Guard.NotNull(this.ColdStakingManager, nameof(this.ColdStakingManager));
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.walletTransactionHandler = walletTransactionHandler;
+        }
+
+        /// <summary>
+        /// Gets general information related to cold staking.
+        /// </summary>
+        /// <param name="request">A <see cref="GetColdStakingInfoRequest"/> object containing the
+        /// parameters  required to obtain cold staking information.</param>
+        /// <returns>A <see cref="GetColdStakingInfoResponse"/> object containing the cold staking information.</returns>
+        [Route("get-cold-staking-info")]
```

Comment:
Can we rename `get-cold-staking-info` to `cold-staking-info`. The get is implied by the HTTP verb.

Reply:
Changed.